### PR TITLE
[AWS] Add dimensions to S3 Storage Lens

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Set fields as dimensions for TSDB support.

The ECS fields were chosen based on https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359.

In addition to that, a new field `dimensions_id` was created and set as dimension. It is a fingerprint of `aws.dimension.*` fields that uniquely identify a metric.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

Relates to https://github.com/elastic/integrations/issues/6293

### CODE REVIEW

File: packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+description: "Ingest Pipeline for Amazon S3 Storage Lens metrics"
+
+processors:
+  - rename:
```
Comment (tetianakravchenko) [inline packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml:5]: from description: ` It is a fingerprint of aws.dimension.* fields that uniquely identify a metric.`
shouldn't it be fingerprint processor then?

File: packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      fields: ["aws.dimensions"]
+      target_field: ["aws.s3_storage_lens.dimensions_id"]
+      ignore_missing: true
+
```
Comment (tetianakravchenko) [inline packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml:9]: please remove the empty lines

File: packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+description: "Ingest Pipeline for Amazon S3 Storage Lens metrics"
+
+processors:
+  - rename:
+      fields: ["aws.dimensions"]
+      target_field: ["aws.s3_storage_lens.dimensions_id"]
+      ignore_missing: true
```
Comment (tetianakravchenko) [inline packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml:8]: is it possible that this field is missing? should we add `ignore_missing`?

Comment (tetianakravchenko) [reply packages/aws/data_stream/s3_storage_lens/elasticsearch/ingest_pipeline/default.yml:6]: oh, sorry, I meant `ignore_failure`, but I see you've added `on_failure`

**Review by tetianakravchenko — APPROVED** [review]

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**